### PR TITLE
refactor(signer): Add enum for version-specific data to `BlockInfo`

### LINKED ISSUE #4816: [stacks-signer] seperate common state for v0 and v1 block info out and wrap specific internal state in an enum
Context https://github.com/stacks-network/stacks-core/pull/4788#discussion_r1607006217

### PR DESCRIPTION
**Pull request by jbencin**
### Description

Add enum for version-specific (Signer v0 or v1) data to `BlockInfo`

### Applicable issues

- fixes #4816

### Checklist

- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Review by obycode — APPROVED** [review]
LGTM!

File: stacks-signer/src/signerdb.rs [hunk 1]
```diff
@@ -34,6 +34,73 @@ use stacks_common::util::hash::Sha512Trunc256Sum;
 use stacks_common::{debug, error};
 use wsts::net::NonceRequest;
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default)]
+/// Information specific to Signer V1
+pub struct BlockInfoV1 {
+    /// The associated packet nonce request if we have one
+    pub nonce_request: Option<NonceRequest>,
+    /// Whether this block is already being signed over
+    pub signed_over: bool,
```
Comment (kantai) [inline stacks-signer/src/signerdb.rs:43]: `signed_over` is used in `v0` too: it indicates whether or not the local signer has signed off on the block.

File: stacks-signer/src/signerdb.rs [hunk 2]
```diff
@@ -543,6 +604,8 @@ mod tests {
         let db_path = tmp_db_path();
         let mut db = SignerDb::new(db_path).expect("Failed to create signer db");
         let (mut block_info, block_proposal) = create_block();
+        // We'll need the V1 data fields for this
+        block_info.ext = ExtraBlockInfo::V1(BlockInfoV1::default());
```
Comment (kantai) [inline stacks-signer/src/signerdb.rs:608]: `get_first_signer_block_in_tenure` is used in `v0` (as is the `signed_over` field)

**Review by kantai — COMMENTED** [review]
I like the idea of using an enum for version specific data here, but I think this breaks `v0` at the moment: `signed_over` is used to indicate whether or not a block has been signed by the local node. Maybe it would make sense to add a new field to the `v0` enum for this, or maybe the `signed_over` field should just move back into the common struct.

**Review by obycode — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Pull request merged**